### ISSUE TITLE
Bundled Data-Item Transactions should not return for /tx routes

### ISSUE THREAD

**Issue by jim-toth** [textury/arlocal#79]
Going off of Arweave Gateway behavior, bundled data-item transactions shouldn't be fetchable from **_any_** `/tx` routes, but this works with `arlocal` for most of them (e.g. `/tx/[txid]/data`)

I was confused when trying to fetch a bundled data-item on mainnet via `arweave.net/tx/[txid]/data` even though it was fetchable via `arweave.net/[txid]`.  So it seems to me some kind of mechanism to "not find" these data-item tx would need to be implemented.

**Comment by jim-toth** [comment]
@fuadop hey so, after testing this out, i'm no longer able to fetch the top level bundle tx from `/tx/[txid]` endpoints.

To clarify, the top-level tx with binary bundle format _**should**_ return from `/tx/[txid]`, but the data item transactions inside that one should not.

**Comment by jim-toth** [comment]
There's already what seems to be an unused `parent` field in the transactions table, which could be used to filter out data items.

**Comment by fuadop** [comment]
Hey @jim-toth 👋,

On mainnet I can't resolve bundles with `/tx/[txid]` endpoint.

**Comment by jim-toth** [comment]
Try this one `467_FKC1i8RioZf4HcdWe8JLSMTCSForWBcyNFdL5GI` https://viewblock.io/arweave/tx/467_FKC1i8RioZf4HcdWe8JLSMTCSForWBcyNFdL5GI
https://arweave.net/tx/467_FKC1i8RioZf4HcdWe8JLSMTCSForWBcyNFdL5GI

On Viewblock, they cache tx like gateways do, inside this bundle are 4 `data-item`.  these txids below shouldn't be fetchable but the entire bundle tx as a whole is.

data item txids from the above bundle:
[16wsPZqNMLFstBMkS0FJv3-LfUHs_yb2et_pcUTYfe0](https://viewblock.io/arweave/tx/16wsPZqNMLFstBMkS0FJv3-LfUHs_yb2et_pcUTYfe0)
[6WKRQh1Dhw3KOG85yhRUFOFJFySQGNN0mgEyCpnSj0w](https://viewblock.io/arweave/tx/6WKRQh1Dhw3KOG85yhRUFOFJFySQGNN0mgEyCpnSj0w)
[A17qjWSod9sgVR9yl68Xqqu7TTBXCjDx0C3uU5SEfSQ](https://viewblock.io/arweave/tx/A17qjWSod9sgVR9yl68Xqqu7TTBXCjDx0C3uU5SEfSQ)
[kdUDTu8pZobjusMJSn-zVJs-PMKR_-S_RO-PwG-Xn-s](https://viewblock.io/arweave/tx/kdUDTu8pZobjusMJSn-zVJs-PMKR_-S_RO-PwG-Xn-s)

**Comment by fuadop** [comment]
Cool @jim-toth , so looking at the view block txs the one which should be accessible has bundle format and bundle version tags but the nested ones don't have. How do I differentiate them?🤔

**Comment by fuadop** [comment]
Hey @jim-toth , I think I understand now 🤧. The main bundle should be accessible but the txs in the bundle shouldn't be accessible. Right?

**Comment by jim-toth** [comment]
> The main bundle should be accessible but the txs in the bundle shouldn't be accessible. Right?

yup, sorry for the confusing terminology!

I believe Gateways use a prop on data-item tx called `bundledIn` (might be latest convention) or `parent`, which could be passed to the recurse call in `txPostRoute()` instead of `txInBundle: true`

https://github.com/textury/arlocal/blob/main/src/routes/transaction.ts#L198

```typescript
await txPostRoute({
  ...ctx,
  connection: ctx.connection,
  dbPath: ctx.dbPath,
  logging: ctx.logging,
  network: ctx.network,
  request: {
    ...ctx.request,
    body: {
      id: bundle.getIdBy(i),
      bundledIn: data.id,
      ...item.toJSON(),
    },
  },
  // @ts-ignore
  txInBundle: true, // remove this and check for request.body.bundledIn later in the code instead
});
```

then in your tx middleware, filter out tx based on `bundledIn` not being `NULL`

**Comment by fuadop** [comment]
Hey @jim-toth 👋🏾 , 

Thanks for the idea 😊 . Can you please test on this branch for the main time `bg/restrict-data-items`, to verify it works fine.

**Comment by fuadop** [comment]
A new version has been released @jim-toth  (1.1.27).

**Comment by jim-toth** [comment]
Thanks @fuadop !
